### ISSUE TITLE
Stamina continues to deplete even after movement has stopped

### ISSUE THREAD

**Issue by imakeinternet** [suriyun-production/mmorpg-kit-docs#211]
Toggling run mode with SHIFT causes stamina to deplete even if you are standing still. Toggling run mode off stops the depletion but this should happen automatically once the character has stopped moving.

unity 2019.3.0f6 / kit 1.52c / demo mmo project / dedicated server

**Comment by ben-roberts** [comment]
I think this is happening as the only check on depleting stamina is whether the sprint toggle has been set. Possible fix would be to check MovementState!=None as well as that flag in 

```public override float GetDecreasingStaminaPerSeconds(BaseCharacterEntity character)
        {
            if (!character.ExtraMovementState.HasFlag(ExtraMovementState.IsSprinting))
                return 0f;
            return staminaDecreasePerSeconds;
        }```

**Comment by insthync** [comment]
This was fixed in 1.53. If this issue still occurs, you can reopen this issue post.
